### PR TITLE
Feature/154 improve user experience

### LINKED ISSUE #154: Improve user experience
### Motivation

Improves user experience while running generation tool. 
- Clears up the confusion between models compatibility and used SDK.
- Notifies user of using invalid parameters

#### 1. Improve logging of intended model SDK with version
actual message:
```
16 content type models were successfully created.
CustomTypeProvider class was successfully created. 
```
proposed message:
```
16 content type models were successfully created.
CustomTypeProvider class was successfully created. 
Models were generated for Delivery/Management SDK version x.x.x
```

#### 2. Introduce check for invalid parameters instead of ignoring them
We have 2 options for reacting to invalid character:
  a. Fail with an error message (breaking change)
  b. Display error message and continue with execution if possible

### Additional context
Discovered while [https://github.com/kontent-ai/management-sdk-net/issues/183](https://github.com/kontent-ai/management-sdk-net/issues/183)

### PR DESCRIPTION
**Pull request by Sevitas**
### Motivation

Which issue does this fix? Fixes #154 

### Checklist

- [x] Code follows coding conventions held in this repo
- [x] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

Check if all paramers work as expected, invalid stops execution and return a message, version of target sdk is printed to the output console.

### CODE REVIEW

File: src/Kontent.Ai.ModelGenerator/Program.cs [hunk 1]
```diff
@@ -68,40 +72,10 @@ public static async Task<int> Main(string[] args)
             }
         }
 
-        internal static IDictionary<string, string> GetSwitchMappings(string[] args)
+        private static void PrintSdkVersion(bool managementApi)
```
Comment (Simply007) [inline src/Kontent.Ai.ModelGenerator/Program.cs:75]: I might use `CodeGeneratorOptions` as an argument (if we one time have i.t. GraphQL models, ...)

Comment (Simply007) [reply src/Kontent.Ai.ModelGenerator/Program.cs:75]: And MAYBE we can have PrintSdkVersion as an Extension method, but this is just an idea.

File: src/Kontent.Ai.ModelGenerator/ArgHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        return GeneralMappings
+            .Union(ContainsManageApiArg() ? ManagementMappings : DeliveryMappings)
+            .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+
+        bool ContainsManageApiArg() =>
+            args.Where((value, index) => (value is "-m" or "--managementapi") && index + 1 < args.Length && args[index + 1] == "true").Any();
+    }
+
+    public static bool ContainsUnsupportedArg(string[] args)
```
Comment (Simply007) [inline src/Kontent.Ai.ModelGenerator/ArgHelpers.cs:52]: I am not a fan of inverted conditions/method names. 

`ContainValidArgs` might be better  - just and idea, not requirement.

File: test/Kontent.Ai.ModelGenerator.Tests/ArgHelpersTests.cs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                "true"
+        });
+
+        Assert.Equal(ExpectedManagementMappings, mappings);
+    }
+
+    [Fact]
+    public void ContainsContainsUnsupportedArg_SupportedDeliveryOptions_ReturnsFalse()
+    {
+        var args = AppendValuesToArgs(ExpectedDeliveryMappings)
```
Comment (Simply007) [inline test/Kontent.Ai.ModelGenerator.Tests/ArgHelpersTests.cs:93]: Kinda hard to check with a list of parameters - not sure how to rewrite it.

**Review by Simply007 — COMMENTED** [review]
Just a couple of notes. Overall it is fine.

I might vote to add a tracking header -https://github.com/kontent-ai/.github/blob/main/docs/README.md#analytics-1 => do we want to add this as a part of the PR, or separately?

File: src/Kontent.Ai.ModelGenerator/Program.cs [hunk 4]
```diff
@@ -24,11 +20,17 @@ public static async Task<int> Main(string[] args)
                 // Create an instance of a DI container
                 var services = new ServiceCollection();
 
+                if (ArgHelpers.ContainsUnsupportedArg(args))
+                {
+                    Console.WriteLine("Failed to run due to invalid configuration.");
```
[the remark this answers is not in the document]
Comment (Sevitas) [reply src/Kontent.Ai.ModelGenerator/Program.cs]: Do we want to use it just here, or on other places as well?

File: src/Kontent.Ai.ModelGenerator/ArgHelpers.cs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Kontent.Ai.Delivery.Abstractions;
+using Kontent.Ai.Management.Configuration;
+using Kontent.Ai.ModelGenerator.Core.Configuration;
+
+namespace Kontent.Ai.ModelGenerator;
+
+public static class ArgHelpers
```
Comment (Simply007) [inline src/Kontent.Ai.ModelGenerator/ArgHelpers.cs:11]: We can do it internal right?

File: test/Kontent.Ai.ModelGenerator.Tests/ArgHelpersTests.cs (hunk 3 above)
Comment (Simply007) [reply test/Kontent.Ai.ModelGenerator.Tests/ArgHelpersTests.cs:93]: Haven't found better solution

File: src/Kontent.Ai.ModelGenerator/UsedSdkInfo.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+namespace Kontent.Ai.ModelGenerator;
+
+public class UsedSdkInfo
+{
+    public string Name { get; }
+    public string Version { get; }
+
+    public UsedSdkInfo(Type type, string name)
+    {
+        Version = type == null ? throw new ArgumentNullException(nameof(type)) : Assembly.GetAssembly(type).GetName().Version.ToString(3);
```
Comment (Simply007) [inline src/Kontent.Ai.ModelGenerator/UsedSdkInfo.cs:13]: What does ToString(3) do?

**Review by Simply007 — CHANGES_REQUESTED** [review]
Just minor changes - everything else is OK

File: src/Kontent.Ai.ModelGenerator/UsedSdkInfo.cs (hunk 6 above)
Comment (Sevitas) [reply src/Kontent.Ai.ModelGenerator/UsedSdkInfo.cs:13]: It does print the semantic version (MAJOR.MINOR.PATCH) of the assembly.

File: src/Kontent.Ai.ModelGenerator/Program.cs (hunk 1 above)
Comment (Sevitas) [reply src/Kontent.Ai.ModelGenerator/Program.cs]: Cannot be done in a clear way, so we have decided to keep it here for now

File: src/Kontent.Ai.ModelGenerator/ArgHelpers.cs (hunk 5 above)
Comment (Sevitas) [reply src/Kontent.Ai.ModelGenerator/ArgHelpers.cs]: https://github.com/kontent-ai/model-generator-net/blob/feature/154_improve_user_experience/src/Kontent.Ai.ModelGenerator/ArgHelpers.cs#L11

File: src/Kontent.Ai.ModelGenerator/ArgHelpers.cs (hunk 2 above)
Comment (Sevitas) [reply src/Kontent.Ai.ModelGenerator/ArgHelpers.cs]: https://github.com/kontent-ai/model-generator-net/blob/feature/154_improve_user_experience/src/Kontent.Ai.ModelGenerator/ArgHelpers.cs#L52

File: src/Kontent.Ai.ModelGenerator/Program.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sevitas) [reply src/Kontent.Ai.ModelGenerator/Program.cs]: - https://github.com/kontent-ai/model-generator-net/blob/feature/154_improve_user_experience/src/Kontent.Ai.ModelGenerator/Program.cs#L75
- https://github.com/kontent-ai/model-generator-net/blob/feature/154_improve_user_experience/src/Kontent.Ai.ModelGenerator/ArgHelpers.cs#L67

**Review by Simply007 — APPROVED** [review]

**Pull request merged**